### PR TITLE
[Fix] UNO 532 & UNO 579 - Added validator for parameters of the sliderInteraction

### PR DESCRIPTION
**Pull request by zagovorichev**
**Related to**

https://oat-sa.atlassian.net/browse/UNO-532

https://oat-sa.atlassian.net/browse/UNO-579
 
**Added rules that cover these cases**

1. Lower bound can not be bigger than upper bound.
2. Selected expected result is not in the range.
 
**Steps to reproduce  (for bugs)**

- Create slider
- Set answer
- Adjust range below set answer
- Save

You now have a slider item that cannot be answered by a student, unless they input the answer directly into the html. 

**Test commands**
 
```
cd tao/views/build
npx grunt connect:test qunit:single --test=/taoQtiItem/views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.html
```

**Or in the browser**

http://tao.loc/taoQtiItem/views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.html?_ijt=r9idbuut90rbme8etulrsmnk0n

### CODE REVIEW

File: views/js/qtiCreator/widgets/interactions/sliderInteraction/Helper.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        responseManager: function responseManager(interaction, response, currentResponse) {
+            const lowerBound = interaction.attributes.lowerBound;
+            const upperBound = interaction.attributes.upperBound;
+
+            let validResponse = 0;
+            let errMessage = '';
+            let warnQuestion = '';
+
+            // if response is bigger than allowed
+            if (lowerBound > upperBound) {
```
Comment (habemuscode) [inline views/js/qtiCreator/widgets/interactions/sliderInteraction/Helper.js:41]: Maybe its more cleaner to use a Switch

```
switch (true) {
  case (val < 1000): /* do something */ break;
  case (val < 2000): /* do something */ break;
  ...
  case (val < 30000): /* do something */ break;
}
```

More examples: https://stackoverflow.com/questions/6665997/switch-statement-for-greater-than-less-than

File: views/js/qtiCreator/widgets/interactions/sliderInteraction/Widget.js [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
 ], function(Widget, states){
-    
-    var SliderInteractionWidget = Widget.clone();
-    
+
+    const SliderInteractionWidget = Widget.clone();
+
     SliderInteractionWidget.initCreator = function(){
-        this.registerStates(states);
+        this.registerStates(states)
```
Comment (habemuscode) [inline views/js/qtiCreator/widgets/interactions/sliderInteraction/Widget.js:27]: Semicolon

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        sliderInteractionHelper.responseManager(interaction, response, 5);
+    });
+
+    QUnit.test('Current response is lower than lower bound', function (assert) {
+        assert.expect(1);
+        const ready = assert.async();
+        dialogConfirm.off('message').on('message', function (msg) {
+            assert.equal(msg, 'Change response ["1"] to the lower bound value "2".', 'Dialog has correct message');
+            ready();
+        })
```
Comment (habemuscode) [inline views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js:121]: semicolon

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        sliderInteractionHelper.responseManager(interaction, testResponse, 2);
+    });
+
+    QUnit.test('Current response is bigger than upper bound', function (assert) {
+        assert.expect(1);
+        const ready = assert.async();
+        dialogConfirm.off('message').on('message', function (msg) {
+            assert.equal(msg, 'Change response ["5"] to the upper bound value "3".', 'Dialog has correct message');
+            ready();
+        })
```
Comment (habemuscode) [inline views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js:108]: semicolon

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        sliderInteractionHelper.responseManager(interaction, response, 0);
+    });
+
+    QUnit.test('Check only possible response failed', function (assert) {
+        assert.expect(1);
+        const ready = assert.async();
+        dialogConfirm.off('message').on('message', function (msg) {
+            assert.equal(msg, 'Change response ["0"] to the only possible value "2".', 'Dialog has correct message');
+            ready();
+        })
```
Comment (habemuscode) [inline views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js:81]: semicolon

File: views/js/qtiCreator/widgets/interactions/sliderInteraction/Helper.js (hunk 1 above)
Comment (zagovorichev) [reply views/js/qtiCreator/widgets/interactions/sliderInteraction/Helper.js:41]: 1. `if` faster
2. I thought about the `steps`, what if we need more conditions here? then switch not so clear

File: views/js/qtiCreator/widgets/interactions/sliderInteraction/Widget.js (hunk 2 above)
Comment (zagovorichev) [reply views/js/qtiCreator/widgets/interactions/sliderInteraction/Widget.js:27]: good catch, fixed

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js (hunk 3 above)
Comment (zagovorichev) [reply views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js:121]: fixed

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js (hunk 4 above)
Comment (zagovorichev) [reply views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js:108]: fixed

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js (hunk 5 above)
Comment (zagovorichev) [reply views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js:81]: fixed

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    const interaction = {
+        attributes: {
+            lowerBound: 0,
+            upperBound: 0,
+            step: 1,
+        }
+    }
+
+    const response = {
+        setCorrect: function setCorrect(correct) {
```
[the remark this answers is not in the document]
Comment (zagovorichev) [reply views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js:40]: not sure that understood that commentary, but made it in one row

**Comment by zagovorichev** [comment]
@habemuscode Updated

File: views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (habemuscode) [reply views/js/test/qtiCreator/widgets/interactions/sliderInteraction/Helper/test.js]: I mean, indentation, but now looks fine :)

**Comment by habemuscode** [comment]
@zagovorichev

imagen

**Review by habemuscode — APPROVED** [review]
- [x]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Comment by zagovorichev** [comment]
@habemuscode @AndreyNegrov I've changed component and implemented task UNO-579 here

**Comment by habemuscode** [comment]
Looks fixed UNO-579

imagen

**Review by habemuscode — APPROVED** [review]

**Review by AndreyNegrov — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**